pkg/hubble/parser/seven/http_test.go

```diff
@@ -189,6 +189,54 @@ func TestDecodeL7HTTPRequestRemoveUrlQuery(t *testing.T) {
 	}, f.GetL7().GetHttp())
 }
 
+func TestDecodeL7HTTPRequestRedactHeaders(t *testing.T) {
+	requestPath, err := url.Parse("http://myhost/some/path?foo=bar")
+	require.NoError(t, err)
+	lr := &accesslog.LogRecord{
+		Type:                accesslog.TypeRequest,
+		Timestamp:           fakeTimestamp,
+		NodeAddressInfo:     fakeNodeInfo,
+		ObservationPoint:    accesslog.Ingress,
+		SourceEndpoint:      fakeSourceEndpoint,
+		DestinationEndpoint: fakeDestinationEndpoint,
+		IPVersion:           accesslog.VersionIPv4,
+		Verdict:             accesslog.VerdictForwarded,
+		TransportProtocol:   accesslog.TransportProtocol(u8proto.TCP),
+		ServiceInfo:         nil,
+		DropReason:          nil,
+		HTTP: &accesslog.LogRecordHTTP{
+			Code:     0,
+			Method:   "POST",
+			URL:      requestPath,
+			Protocol: "HTTP/1.1",
+			Headers: http.Header{
+				"Host":        {"myhost"},
+				"Traceparent": {"00-4bf92f3577b34da6a3ce929d0e0e4736-00f067aa0ba902b7-01"},
+			},
+		},
+	}
+	lr.SourceEndpoint.Port = 56789
+	lr.DestinationEndpoint.Port = 80
+
+	opts := []options.Option{options.Redact(nil, []string{"http-headers"})}
+	parser, err := New(log, nil, nil, nil, nil, opts...)
+	require.NoError(t, err)
+
+	f := &flowpb.Flow{}
+	err = parser.Decode(lr, f)
+	require.NoError(t, err)
+	assert.Equal(t, &flowpb.HTTP{
+		Code:     0,
+		Method:   "POST",
+		Url:      "http://myhost/some/path?foo=bar",
+		Protocol: "HTTP/1.1",
+		Headers: []*flowpb.HTTPHeader{
+			{Key: "Host", Value: "HUBBLE_REDACTED"},
```

Comment:
The host header being redacted gives me pause about this approach. I made a comment on the CFP https://github.com/cilium/cilium/issues/23887#issuecomment-1629275365 to discuss the overall approach being taken.